### PR TITLE
Add the 'xor' boolean operator

### PR DESCRIPTION
**Pull request by EmileRolley**
Releated issue : #87 

Add the exclusive or operator.
In the english and abbreviated syntax the keyword is `xor`.
In the french syntax it's `ou bien`.

Changelog:

* New token `XOR`.
* New binop `Xor`.
* In [src/catala/lcalc/to_ocaml.ml](https://github.com/EmileRolley/catala/blob/xor/src/catala/lcalc/to_ocaml.ml#L59) the `Xor` binop is formatted into `<>`. 
* Add tests in [tests/test_bool](https://github.com/EmileRolley/catala/tree/xor/tests/test_bool).
* Update the syntax highlighting by adding `xor` and `ou bien` to operators.

### CODE REVIEW

**Comment by denismerigoux** [comment]
Formatting comment: the `ocamlformat` version to use is `0.15.0`, I suspect you're using `0.16.0`.

**Comment by EmileRolley** [comment]
> Formatting comment: the `ocamlformat` version to use is `0.15.0`, I suspect you're using `0.16.0`.

True :sweat_smile:

File: syntax_highlighting/fr/vscode/syntaxes/catalavs.xml [hunk 1]
```diff
@@ -86,7 +86,7 @@
         </dict>
         <dict>
           <key>match</key>
-          <string>([^`#])</string>
+          <string>([^`])</string>
```
Comment (denismerigoux) [inline syntax_highlighting/fr/vscode/syntaxes/catalavs.xml:89]: Hum, why remove the `#` ? Shouldn't it be `<string>([^`#>])</string>` ?

**Review by denismerigoux — APPROVED** [review]
Thanks Emile for this new wonderful PR! I just added a little comment but it's minor detail. Will merge soon.

File: syntax_highlighting/fr/vscode/syntaxes/catalavs.xml (hunk 1 above)
Comment (EmileRolley) [reply syntax_highlighting/fr/vscode/syntaxes/catalavs.xml]: Little mc I think..

**Comment by denismerigoux** [comment]
Fixes #87

**Pull request merged**